oak_functions/wasm/src/lib.rs

```diff
@@ -36,388 +35,452 @@ use oak_functions_abi::{
 };
 use oak_functions_extension::{ExtensionFactory, OakApiNativeExtension};
 use oak_logger::{Level, OakLogger};
-use wasmi::ValueType;
+use wasmi::{AsContext, AsContextMut, Func, Memory, MemoryType};
 
 const MAIN_FUNCTION_NAME: &str = "main";
 const ALLOC_FUNCTION_NAME: &str = "alloc";
+const MEMORY_NAME: &str = "memory";
 
-/// Wasm host function index numbers for `wasmi` to map import names with. This numbering is not
-/// exposed to the Wasm client. See <https://docs.rs/wasmi/0.6.2/wasmi/trait.Externals.html>
-const READ_REQUEST: usize = 0;
-const WRITE_RESPONSE: usize = 1;
-const INVOKE: usize = 4;
+// Needs to be consistent with definition of the Wasm import module in the Oak Functions ABI.
+const HOST: &str = "oak_functions";
```

Comment:
Still not sure why this is called HOST instead of, say, MODULE_NAME? Just for my understanding (and maybe worth a comment).

Reply:
I followed the example in https://docs.rs/wasmi/latest/wasmi/ where they called it host. I think it is a matter of perspective, but I changed to OAK_FUNCTIONS now.